### DISCUSSION TITLE
Cycling diffs in log buffer with --patch given

### DISCUSSION THREAD

**Discussion by ghost** [magit/magit#4273] [Ideas]
Hello,

I'm currently trying to implement a substitute for `magit-section-cycle-diffs` which doesn't work in log buffers.  But I now wonder: wasn't it possible to toggle hiding diffs in log buffers in the past?  Is it planned to support this?  Or do I miss something?

TIA, Michael.

**Comment by kyleam** [comment]
> But I now wonder: wasn't it possible to toggle hiding diffs in log buffers in the past?

Via `magit-section-cycle-diffs`?  Not as far as I can tell.  Since the time it was introduced (ef079964, 2014-07-29), it doesn't look like that command has dealt with log buffers.

Other cycling commands (`magit-section-toggle`, `magit-section-show-level-*-all`, ...) will toggle the `--patch`-related sections in non-graph log buffers.  When `--graph` is included as an argument, the diff isn't parsed into a proper section.

**Comment by ghost** [comment]
Kyle Meyer <notifications@github.com> writes:

> Other cycling commands (magit-section-toggle,
> magit-section-show-level-*-all, ...) will toggle the --patch-related sections in
> non-graph log buffers. When --graph is included as an argument, the diff isn't
> parsed into a proper section.

Ah, ok, I had used --graph.

I see that the problem that sometimes the wrong section is highlighted
when moving around in the buffer after hiding diffs still exists.  It
has something to do with stickiness of the invisible property.  AFAIR
the case when overlays are used to make text invisible is the harder
one.  Do you remember that problem and if it had been tried to fix it?

**Comment by ghost** [comment]
Michael Heerdegen <michael_heerdegen@web.de> writes:

> I see that the problem that sometimes the wrong section is highlighted
> when moving around in the buffer after hiding diffs still exists.  It
> has something to do with stickiness of the invisible property.  AFAIR
> the case when overlays are used to make text invisible is the harder
> one.  Do you remember that problem and if it had been tried to fix it?

This had been discussed in Emacs Bug#23079.  Two important points were:
Magit is changing text properties in the buffer in some hook,
i.e. modifies the buffer, which seems to cause trouble.  I don't recall
the actual hook function.

And Jonas planned to replace the visibility toggling mechanism with
something based on a cache.
